api/src/test/java/org/openmrs/api/ConceptServiceTest.java

```diff
@@ -185,7 +186,201 @@ public void getConceptByName_shouldGetConceptByPartialName() {
 		List<Concept> firstConceptsByPartialNameList = conceptService.getConceptsByName(partialNameToFetch);
 		assertThat(firstConceptsByPartialNameList, containsInAnyOrder(hasId(1), hasId(2)));
 	}
+
+	/**
+	 * @see ConceptService#getPrevConcept(Concept) 
+	 */
+	@Test
+	public void getPrevConcept_shouldReturnPreviousConceptBasedOnConceptId() {
+		executeDataSet(INITIAL_CONCEPTS_XML);
+
+		Concept currentConcept = conceptService.getConcept(4);
+		assertNotNull(currentConcept);
+
+		Concept previousConcept = conceptService.getPrevConcept(currentConcept);
+
+		assertNotNull(previousConcept);
+		assertEquals((Integer)(currentConcept.getConceptId() - 1), previousConcept.getConceptId());
+	}
+
+	/**
+	 * @see ConceptService#getPrevConcept(Concept)
+	 */
+	@Test
+	public void getPrevConcept_shouldReturnNullIfNoPrevConceptId() {
+		executeDataSet(INITIAL_CONCEPTS_XML);
+
+		Concept currentConcept = conceptService.getConcept(1);
+		assertNotNull(currentConcept);
+
+		Concept previousConcept = conceptService.getPrevConcept(currentConcept);
+
+		assertNull(previousConcept);
+	}
+
+	/**
+	 * @see ConceptService#getNextConcept(Concept) 
+	 */
+	@Test
+	public void getNextConcept_shouldReturnNextConceptBasedOnConceptId() {
+		executeDataSet(INITIAL_CONCEPTS_XML);
+
+		Concept currentConcept = conceptService.getConcept(3);
+		assertNotNull(currentConcept);
+
+		Concept nextConcept = conceptService.getNextConcept(currentConcept);
+
+		assertNotNull(nextConcept);
+		assertEquals((Integer)(currentConcept.getConceptId() + 1), nextConcept.getConceptId());
+	}
+
+	/**
+	 * @see ConceptService#getNextConcept(Concept)
+	 */
+	@Test
+	public void getNextConcept_shouldReturnNullIfNoNextConceptId() {
+		executeDataSet(INITIAL_CONCEPTS_XML);
+
+		// Use the highest concept ID in your dataset
+		Concept currentConcept = conceptService.getConcept(5497);
+		assertNotNull(currentConcept);
+
+		Concept nextConcept = conceptService.getNextConcept(currentConcept);
+
+		assertNull(nextConcept);
+	}
+
+	/**
+	 * @see ConceptService#getAllConceptProposals(boolean)  
+	 */
+	@Test
+	public void getAllConceptProposals_whenIncludeCompletedIsFalse_shouldReturnOnlyUncompletedProposals() {
+		executeDataSet(INITIAL_CONCEPTS_XML);
+
+		List<ConceptProposal> proposals = conceptService.getAllConceptProposals(false);
+
+		ConceptProposal previousProposal = null;
+		for (ConceptProposal proposal : proposals) {
+			assertEquals(OpenmrsConstants.CONCEPT_PROPOSAL_UNMAPPED, proposal.getState());
+
+			if (previousProposal != null) {
+				assertTrue(previousProposal.getOriginalText().compareTo(proposal.getOriginalText()) <= 0);
+			}
+			previousProposal = proposal;
+		}
+	}
+
+	/**
+	 * @see ConceptService#getConceptProposals(String) )
+	 */
+	@Test
+	public void getConceptProposals_shouldReturnProposalsMatchingTextAndUnmappedState() {
```

Comment:
There are some unnecessary spaces in this method

Reply:
Removed